File: labanoro-draugai-bl/src/main/java/lt/virai/labanoroDraugai/bl/services/TransactionService.java
package lt.virai.labanoroDraugai.bl.services;

import lt.virai.labanoroDraugai.bl.exceptions.LabanoroException;
import lt.virai.labanoroDraugai.domain.entities.PointPurchase;

import java.time.LocalDate;
import java.util.List;

/**
 * Created by Žilvinas on 2016-05-10.
 */
public interface TransactionService {
    PointPurchase createPurchase(Integer userId, Integer amount);

    List<PointPurchase> getAllPurchases();

    void confirmPurchase(int purchaseId);

    void rejectPurchase(int purchaseId);

    boolean hasUserPaidAnnualPayment(int userId);

    LocalDate getLastAnnualPaymentDate(int userId);

    void payAnnualPayment(Integer userId) throws LabanoroException;
}

File: labanoro-draugai-ui/src/main/webapp/public/js/services/transactionService.js
app.service('transactionService', ['$http', function ($http) {
    var baseUrl = 'rest/purchases/';

    this.createPurchase = function (amount) {
        return $http.post(baseUrl + 'createPurchase', amount);
    };

    this.sendPoints = function (id, amount) {
        return $http.post(baseUrl + 'sendPoints/' + id, amount);
    };

    this.payAnnualPayment = function () {
        return $http.get(baseUrl + 'payAnnualPayment');
    };

    this.getAllPurchases = function () {
        return $http.get(baseUrl + 'getAllPurchases')
            .then(function (result) {
                return result.data;
            }).catch(function (response) {
                console.log(response);
            });
    };

    this.confirmPurchase = function (purchaseId) {
        return $http.post(baseUrl + "confirmPurchase", purchaseId);
    };

    this.rejectPurchase = function (purchaseId) {
        return $http.post(baseUrl + "rejectPurchase", purchaseId);
    };

    this.getAnnualPaymentSize = function () {
        return $http.get(baseUrl + 'getAnnualPaymentSize')
            .then(function (result) {
                return result.data;
            }).catch(function (response) {
                console.log(response);
            });
    };

}]);
